### PR TITLE
Expand AMD accelerators.

### PR DESCRIPTION
**Pull request by haringsrob**
Hey there!

I have added some AMD accelerators to this package.

While I added architecture, it is not yet exposed in the CatalogItem, I think we can add that as well (and add them for Nvidia as well) so these can be pinned?

Let me know.

### CODE REVIEW

**Comment by un-def** [comment]
@haringsrob, thanks for the contribution.

> While I added architecture, it is not yet exposed in the CatalogItem, I think we can add that as well

Please open an issue with your feature proposal, as this change requires some discussion.

File: src/gpuhunt/_internal/constraints.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 from gpuhunt._internal.models import (
     AcceleratorVendor,
+    AmdArchitecture,
```
Comment (un-def) [inline src/gpuhunt/_internal/constraints.py:5]: Please rename it to `AMDArchitecture` for consistent naming (as we already have ALLCAPS `AMDGPUInfo`).

File: src/gpuhunt/_internal/models.py [hunk 2]
```diff
@@ -18,6 +18,18 @@ def bool_loader(x: Union[bool, str]) -> bool:
     return x.lower() == "true"
 
 
+class AmdArchitecture(enum.Enum):
+    CDNA = "CDNA"
+    CDNA2 = "CDNA2"
+    CDNA3 = "CDNA3"
+
+    @classmethod
+    def cast(cls, value: Union["AcceleratorVendor", str]) -> "AcceleratorVendor":
```
Comment (un-def) [inline src/gpuhunt/_internal/models.py:27]: Here and below, ~~AcceleratorVendor~~ → AMDArchitecture

File: src/tests/_internal/test_models.py [hunk 3]
```diff
@@ -18,6 +25,13 @@
         pytest.param(1, None, "tpu-v3", "google", "v3", id="one-tpu-vendor-not-set"),
         pytest.param(1, "google", "tpu-v5p", "google", "v5p", id="one-tpu-vendor-is-set"),
         pytest.param(1, AMD, "MI300X", "amd", "MI300X", id="cast-enum-to-string"),
+        pytest.param(1, AMD, "MI300A", "amd", "MI300A", id="cast-enum-to-string"),
```
Comment (un-def) [inline src/tests/_internal/test_models.py:28]: These new test cases seem redundant, they all check (enum) AMD to (string) "amd" casting.

File: src/tests/_internal/test_models.py [hunk 4]
```diff
@@ -49,6 +63,13 @@ def test_raw_catalog_item_gpu_vendor_heuristic(
         pytest.param(1, None, None, NVIDIA, id="one-gpu-no-name"),
         pytest.param(1, None, "v3", NVIDIA, id="one-gpu-with-any-name"),
         pytest.param(1, "amd", "MI300X", AMD, id="cast-string-to-enum"),
+        pytest.param(1, "amd", "MI300A", AMD, id="cast-string-to-enum"),
```
Comment (un-def) [inline src/tests/_internal/test_models.py:66]: Same here

File: src/tests/_internal/test_models.py (hunk 3 above)
Comment (haringsrob) [reply src/tests/_internal/test_models.py:28]: Yea, I realised this myself in the evening 👍

**Comment by haringsrob** [comment]
Thanks @un-def I changed these as requested. Will open an issue regarding architecture.

Still have to check this in a runtime env tho.

File: src/gpuhunt/_internal/models.py [hunk 5]
```diff
@@ -18,6 +18,18 @@ def bool_loader(x: Union[bool, str]) -> bool:
...
 
 
+class AMDArchitecture(enum.Enum):
+    CDNA = "CDNA"
+    CDNA2 = "CDNA2"
+    CDNA3 = "CDNA3"
+
+    @classmethod
+    def cast(cls, value: Union["AcceleratorVendor", str]) -> "AcceleratorVendor":
+        if isinstance(value, AcceleratorVendor):
```
Comment (un-def) [inline src/gpuhunt/_internal/models.py:28]:
```suggestion
    def cast(cls, value: Union["AMDArchitecture", str]) -> "AMDArchitecture":
        if isinstance(value, AMDArchitecture):
```

**Pull request merged**